src/util/quality_value.rs

```diff
@@ -33,21 +32,18 @@ impl Default for Quality {
 
 /// Represents an item with a quality value as defined in
 /// [RFC7231](https://tools.ietf.org/html/rfc7231#section-5.3.1).
-#[derive(Clone, PartialEq, Debug)]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct QualityValue<T> {
```

Comment:
I've felt for a while now that the `QualityValue` design is kind of annoying to use, but I've never sat down to come up with a better one.

Reply:
Can't say I have any better idea. The API looks ok to me, at least there isn't anything that should be clearly removed